c10/core/StorageImpl.h

```diff
@@ -118,12 +119,12 @@ struct C10_API StorageImpl : public c10::intrusive_ptr_target {
     return resizable_;
   }
 
-  at::DataPtr& mutable_data_ptr() {
-    maybe_materialize_cow();
-    return data_ptr_;
-  }
+  const at::DataPtr& data_ptr() const;
```

Comment:
mehhh, you sure you don't want this inlining? What would be more normal is to do an inlinable conditional, where the slow path goes into a non-inlined function

Reply:
I didn't realize that putting things in a header automatically inlined them without the inline keyword. I moved this around so that I could develop faster without a bunch of recompiles from including this header everywhere. I'll put them back

Reply:
#til https://en.cppreference.com/w/cpp/language/inline#:~:text=A%20function%20defined%20entirely%20inside,(since%20C%2B%2B20).